src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerTextView.cs

```diff
@@ -38,6 +35,9 @@ internal partial class DebuggerTextView : IWpfTextView, IDebuggerTextView, IText
             _debuggerTextLinesOpt = debuggerTextLinesOpt;
             BufferGraph = bufferGraph;
             IsImmediateWindow = isImmediateWindow;
+
+            // The editor requires the current top buffer.
+            _innerTextView.Properties.AddProperty(CompletionRoot, bufferGraph.TopBuffer);
```

Comment:
i can't figure out what this is for.  Who reads this value out?  What does it mean that "the editor requires the current top buffer"?

Reply:
Here is how Editor explained this:
               // In certain scenarios, TextView.TextSnapshot is not the appropriate snapshot to use.
                // For example, in the watch window (C# debugger), TextView.TextSnapshot corresponds to the single line for the expression.
                // Roslyn uses the property bag to indicate the correct snapshot to use.
 

---
In reply to: [232508037](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508037) [](ancestors = 232508037)

Reply:
Who reads the value out?  where does that happen?  thanks!

Reply:
Tagging @jasonmalinowski Something seems weird here with projections.  i feel like this shouldn't be something completion needs to know about.

Reply:
The editor reads the value. They do this on each OpenOrUpdate of a session.

Here is how Editor explained this:

csharp
// In certain scenarios, TextView.TextSnapshot is not the appropriate snapshot to use.
// For example, in the watch window (C# debugger), TextView.TextSnapshot corresponds to the single line for the expression.
// Roslyn uses the property bag to indicate the correct snapshot to use.

---
In reply to: [232894787](https://github.com/dotnet/roslyn/pull/29016#discussion_r232894787) [](ancestors = 232894787)

Reply:
To be tracked as [https://github.com/dotnet/roslyn/issues/31189](https://github.com/dotnet/roslyn/issues/31189)
Not blocking to merge under a feature flag. Should be resolved before removing the feature flag.

---
In reply to: [232894832](https://github.com/dotnet/roslyn/pull/29016#discussion_r232894832) [](ancestors = 232894832)

Reply:
Likely, debugger tests will fail. I'm not sure we have tests covering completion + debugger at the same time.

---
In reply to: [233083182](https://github.com/dotnet/roslyn/pull/29016#discussion_r233083182) [](ancestors = 233083182)